monai/transforms/utility/array.py

```diff
@@ -1715,3 +1717,135 @@ def __call__(self, img: NdarrayOrTensor, meta_dict: Mapping | None = None) -> Nd
         if self._do_transform:
             img = self.filter(img)
         return img
+
+
+class ApplyTransformToPoints(InvertibleTransform, Transform):
+    """
+    Transform points between image coordinates and world coordinates.
+    The input coordinates are assumed to be in the shape (C, N, 2 or 3), where C represents the number of channels
+    and N denotes the number of points. It will return a tensor with the same shape as the input.
+
+    Args:
+        dtype: The desired data type for the output.
+        affine: A 3x3 or 4x4 affine transformation matrix applied to points. This matrix typically originates
+            from the image. For 2D points, a 3x3 matrix can be provided, avoiding the need to add an unnecessary
+            Z dimension. While a 4x4 matrix is required for 3D transformations, it's important to note that when
+            applying a 4x4 matrix to 2D points, the additional dimensions are handled accordingly.
+            The matrix is always converted to float64 for computation, which can be computationally
+            expensive when applied to a large number of points.
+            If None, will try to use the affine matrix from the input data.
+        invert_affine: Whether to invert the affine transformation matrix applied to the points. Defaults to ``True``.
+            Typically, the affine matrix is derived from an image and represents its location in world space,
+            while the points are in world coordinates. A value of ``True`` represents transforming these
+            world space coordinates to the image's coordinate space, and ``False`` the inverse of this operation.
+        affine_lps_to_ras: Defaults to ``False``. Set this to ``True`` if all of the following are true:
+            1) The image is read by `ITKReader`,
+            2) The `ITKReader` has `affine_lps_to_ras=True`,
+            3) The data is in world coordinates.
+            This ensures the correct application of the affine transformation between LPS (left-posterior-superior)
+            and RAS (right-anterior-superior) coordinate systems. This argument ensures the points and the affine
+            matrix are in the same coordinate system.
+
```

Comment:
Hey @KumoLiu 
So I tested this aspect of the code. Here is what I did. Instead of using `ITKReader` I used `NibabelReader` for loading the images. As I used a different reader I changed the `apply_lps_to_ras` to `False` as indicated in the docstring. 
I get the following result as transformed points 
```
metatensor([[[1001.9410,  -36.8047,   78.0000],
         [1028.0130,  -32.9858,   78.0000],
         [1059.5454,  -50.9436,   78.0000],
         [1087.0344,  -61.4492,   78.0000],
         [1063.8782, -132.3439,   78.0000],
         [ 996.8077, -168.7347,   78.0000],
         [1011.6713, -141.4345,   78.0000],
         [1032.8000, -110.6216,   78.0000],
         [1020.5166,  -78.3613,   78.0000],
         [ 999.7059,  -41.3652,   78.0000]]], dtype=torch.float64)
```
You can try the same by changing my gist. However, if I change the `apply_lps_to_ras` as `True` with `NibabelReader`.

Reply:
> Hey @KumoLiu So I tested this aspect of the code. Here is what I did. Instead of using `ITKReader` I used `NibabelReader` for loading the images. As I used a different reader I changed the `apply_lps_to_ras` to `False` as indicated in the docstring.

Hi @vikashg, thanks for pointing this out. Let me clarify, this argument should set to true when the input point data is annotated in the RAS coordinate. I just updated the docstring, please help take a look again to see whether it's clearer now.